File: java/client/src/main/java/com/github/fernthedev/lightchat/client/ILogManager.java
package com.github.fernthedev.lightchat.client;

import com.github.fernthedev.lightchat.core.api.APIUsage;

@APIUsage
@SuppressWarnings("unused")
public interface ILogManager {

    void log(String log);

    void logError(String log, Throwable e);

    void info(String s);

    void debug(String s);

    void error(String s);
}

File: java/client/src/main/java/com/github/fernthedev/lightchat/client/ILogManager.kt
package com.github.fernthedev.lightchat.client

import com.github.fernthedev.lightchat.core.api.APIUsage

@APIUsage
@Suppress("unused")
interface ILogManager {
    fun log(log: String?)
    fun logError(log: String?, e: Throwable?)
    fun info(s: String?)
    fun debug(s: String)
    fun error(s: String?)
}